src/Controller/Rest/Exception/InvalidPayloadException.php

```diff
@@ -0,0 +1,23 @@
+<?php
+declare(strict_types=1);
+
+namespace YapepBase\Controller\Rest\Exception;
+
+class InvalidPayloadException extends Exception
+{
+    public function __construct(string $jsonDecodeError)
+    {
+        $message = 'Failed to decode the request as valid JSON. JSON decode error: ' . $jsonDecodeError;
+        parent::__construct($message);
+    }
+
+    public function getRecommendedHttpStatusCode(): int
+    {
+        return 400;
```

Comment:
This should be 422

Reply:
Just an example, this is called HTTP_UNPROCESSABLE_ENTITY.
The names are usually correct and meaningful, you will just always need to open the Httpstatuscodes class and find it first ;)